### DISCUSSION TITLE
Bring back support for commonjs?

### DISCUSSION THREAD

**Discussion by jasperverbeet** [wevm/wagmi#1555] [Q&A]
In my opinion, Wagmi dropped support for commonjs too soon. 

I've been using Wagmi successfully in my Nest.js/Next.js monorepo project for a while now, but I'm stuck on an older version of Wagmi because it lacks CommonJS support in newer versions. We also can't migrate to ESM for Node since Jest doesn't fully support all the features we're using.

What was the reasoning behind removing CommonJS support, and is there a chance of it coming back?

**Comment by quankuiyu** [comment]
I'd like to ask you a question I'm using different chains

> network does not support ENS (operation="lookupAddress", network="sepolia", code=UNSUPPORTED_OPERATION, version=providers/5.7.2)
How should this error be resolved

const { chains, provider, webSocketProvider } = configureChains(
  [mainnet, sepolia],
  [publicProvider()],
)

**Comment by tmm** [comment]
This was a hard choice to make, but ultimately think its in the best interest of moving wagmi forward and aligning with the future of the web. Some reasons we dropped support:
- [Dual package hazard](https://github.com/GeoffreyBooth/dual-package-hazard)
- Most modern tools support ESM-only with little to no extra configuration
- Native ESM support has been out for a while now ([since 2019](https://blog.logrocket.com/es-modules-in-node-js-12-from-experimental-to-release/))

What version of Next.js are you using? Full support for ESM [landed in Next.js 12](https://nextjs.org/blog/next-12#es-modules-support-and-url-imports) with experimental support in Next.js 11. If you aren't able to get Jest to work with its [experimental support](https://jestjs.io/docs/ecmascript-modules), then [Vitest](https://vitest.dev) is a solid API-compatible alternative.

**Reply by jasperverbeet** [reply]
Thank you for your response and for explaining the reasoning behind the decision to remove CommonJS support in Wagmi. I understand that these types of decisions are often difficult to make, and I appreciate the thought that goes into them.

Our issue with migrating to ESM relates to Nest.js, which is not fully compatible with ESM in our case. We have been unable to upgrade due to compatibility issues with Jest. While I understand that ESM is the future of the web and that adopting it can bring many benefits, in our situation we are unable to make the transition at this time. However, we will look into using Vitest as an alternative testing framework, as you suggested.

I do understand your standpoint and the need to move Wagmi forward, and I will continue to follow the development of the project and consider any future updates. Thank you again for your time and for considering my feedback.
